carina-api/src/main/java/com/qaprosoft/carina/core/foundation/api/AbstractApiMethodV2.java

```diff
@@ -219,20 +217,39 @@ public void validateResponse(JSONCompareMode mode, String... validationFlags) {
         }
     }
 
+    public void validateXmlResponse(XmlCompareMode mode) {
+        if (actualRsBody == null) {
+            throw new RuntimeException("Actual response body is null. Please make API call before validation response");
+        }
+        if (rsPath == null) {
+            throw new RuntimeException("Please specify rsPath to make Response body validation");
+        }
+        try {
+            XmlComparator.compare(actualRsBody, rsPath, mode);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
     /**
      * @param validationFlags
      *            parameter that specifies how to validate JSON response. Currently only array validation flag is supported.
      *            Use JsonCompareKeywords.ARRAY_CONTAINS enum value for that
      */
     public void validateResponse(String... validationFlags) {
-        validateResponse(JSONCompareMode.NON_EXTENSIBLE, validationFlags);
+        ContentType contentType = this.getClass().getAnnotation(ContentType.class);
+        if (contentType.type().equals(XML_CONTENT_TYPE.getHeaderValue())) {
```

Comment:
what will happen if ContentType annotation is missing?

Reply:
I think it's better to reuse if() from validateResponseAgainstSchema() method to handle all cases.